src/test/scala/is/hail/expr/ir/IRSuite.scala

```diff
@@ -120,4 +121,17 @@ class IRSuite extends TestNGSuite {
     assertEvalsTo(ArrayFilter(a, "x",
       ApplyComparisonOp(LT(TInt32()), Ref("x", TInt32()), I32(6))), FastIndexedSeq(3))
   }
+
+  @Test def testArrayRange() {
+    val range = { (start: Int, stop: Int, step: Int) => ArrayRange(I32(start), I32(stop), I32(step)) }
+
+    assertEvalsTo(range(-9, -10, 2), FastIndexedSeq())
+    assertEvalsTo(range(-9, -10, 3), FastIndexedSeq())
+    assertEvalsTo(range(1, 5, -2), FastIndexedSeq())
+    assertEvalsTo(range(5, 1, -2), FastIndexedSeq(5, 3))
```

Comment:
You should just expand the cases in the testArrayRange bit where you loop over values for start, stop and step.